src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -935,11 +947,26 @@ private void ParseOption(string name, string value)
                 case "sockettimeoutms":
                     _socketTimeout = ParseTimeSpan(name, value);
                     break;
-                case "ssl":
-                    _ssl = ParseBoolean(name, value);
+                case "ssl": // Obsolete
+                case "tls":
+                    var tlsValue = ParseBoolean(name, value);
+                    if (_tls.HasValue && _tls.Value != tlsValue)
+                    {
+                        throw new MongoConfigurationException("tls has already been configured with a different value.");
+                    }
+                    _tls = tlsValue;
                     break;
-                case "sslverifycertificate":
-                    _sslVerifyCertificate = ParseBoolean(name, value);
+                case "sslverifycertificate": // Obsolete
+                    {
+                        var tlsInsecureValue = !ParseBoolean(name, value);
+                        _tlsInsecure = ParseTlsInsecure(tlsInsecureValue);
```

Comment:
The method being called is not doing any parsing, it is only validating. Can we rename to:

    EnsureTlsInsecureIsValid